Source/SwiftLintBuiltInRules/Rules/Lint/PrivateSwiftUIStatePropertyRule.swift

```diff
@@ -244,7 +74,74 @@ private extension PrivateSwiftUIStatePropertyRule {
                 return
             }
 
-            violations.append(decl.bindingSpecifier.positionAfterSkippingLeadingTrivia)
+            if let accessLevelModifier = decl.modifiers.accessLevelModifier {
+                violations.append(accessLevelModifier.positionAfterSkippingLeadingTrivia)
+            } else {
+                violations.append(decl.bindingSpecifier.positionAfterSkippingLeadingTrivia)
+            }
+        }
+    }
+
+    final class Rewriter: ViolationsSyntaxRewriter<ConfigurationType> {
+        /// LIFO stack that stores type inheritance clauses for each visited node
+        /// The last value is the inheritance clause for the most recently visited node
+        /// A nil value indicates that the node does not provide any inheritance clause
+        private var visitedTypeInheritances = Stack<InheritanceClauseSyntax?>()
+
+        override func visit(_ node: ClassDeclSyntax) -> DeclSyntax {
+            visitedTypeInheritances.push(node.inheritanceClause)
+            return super.visit(node)
+        }
+
+        override func visit(_ node: StructDeclSyntax) -> DeclSyntax {
+            visitedTypeInheritances.push(node.inheritanceClause)
+            return super.visit(node)
+        }
+
+        override func visit(_ node: ActorDeclSyntax) -> DeclSyntax {
+            visitedTypeInheritances.push(node.inheritanceClause)
+            return super.visit(node)
+        }
+
+        override func visitPost(_ node: Syntax) {
+            guard node.is(ClassDeclSyntax.self) ||
+                    node.is(StructDeclSyntax.self) ||
+                    node.is(ActorDeclSyntax.self) else { return }
+            visitedTypeInheritances.pop()
+        }
+
+        override func visit(_ node: VariableDeclSyntax) -> DeclSyntax {
+            guard
+                let parent = node.parent,
+                parent.is(MemberBlockItemSyntax.self),
+                let inheritanceClause = visitedTypeInheritances.peek() as? InheritanceClauseSyntax,
+                inheritanceClause.conformsToApplicableSwiftUIProtocol,
```

Comment:
Would be better to store just this attribute instead of the whole object (also in the visitor).

Reply:
I modified the stack so that it just stores the Bool result of this computed property. `true` tells the visitor or rewriter that the current scope should be evaluated for violations.